packages/router/src/adapters/redisCache/cache.ts

```diff
@@ -0,0 +1,40 @@
+import { BigNumber, constants } from "ethers/lib/ethers";
+
+import {
+  GetOutstandingLiquidityParams,
+  StoreOutstandingLiquidityParams,
+  RemoveOutstandingLiquidityParams,
+} from "../../lib/entities";
+
+import { getRedis } from ".";
+
+export const getOutstandingLiquidity = async (g: GetOutstandingLiquidityParams): Promise<BigNumber> => {
+  const redis = getRedis();
+  const { assetId, chainId } = g;
+
+  // use SCAN to get all the keys for the given assetId and chainId
+  const [, keys] = await redis.scan(0, "match", `outstanding-liquidity:${chainId}:${assetId}*`);
+
+  // use a set to get rid of duplicates, apparently SCAN can return duplicated
+  // https://redis.io/commands/scan#scan-guarantees
+  const amounts = await redis.mget([...new Set(keys)]);
+  const num = amounts.filter((x) => !!x).reduce((acc, amount) => acc.add(BigNumber.from(amount)), constants.Zero);
+  return num;
+};
```

Comment:
How efficient is the `scan` at high load?

Reply:
Redis is built to do this type of stuff. I did a bunch of research to find this strategy, which is basically built around this exact use case of array elements that individually expire.